WPAndroid Integration.md

```diff
@@ -0,0 +1,113 @@
+# WordPress Android Integration Guide
+
+### Summary
+
+WPAndroid by default integrates `react-native-bridge` as a binary dependency that's fetched from a remote maven repo.
+The CI in this repo will deploy a new version each time a commit is pushed to `develop` or an open PR, as well as each time a tag is created.
+This is the only required dependency.
+Whenever a new version is deployed, we can simply update `ext.gutenbergMobileVersion` in [build.gradle](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/build.gradle).
+
+During development, we can use composite builds and dependency substitution to point WPAndroid to our local checkout of `gutenberg-mobile`.
+To do this, we need to copy the [local-builds.gradle-example file](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/local-builds.gradle-example) in WPAndroid renaming it to `local-builds.gradle` and change the `localGutenbergMobilePath` value.
+Building the project as such will make WPAndroid build the source code for `react-native-bridge`, `react-native-aztec` and other sublibraries from the locally checked out folder.
+It'll also expect the metro server to be running to fetch the JS bundle.
+
+- [Work with local gutenberg-mobile checkout](#work-with-local-gutenberg-mobile-checkout)
+- [Deploy a new version of `react-native-bridge` from CI](#deploy-a-new-version-of-react-native-bridge-from-ci)
+- [Manually deploy a new version of `react-native-bridge`](#manually-deploy-a-new-version-of-react-native-bridge)
+- [Update WPAndroid's binary version](#update-wpandroids-binary-version)
+- [How to](#how-to)
+    - [How to test a gutenberg-mobile PR in WPAndroid](how-to-test-a-gutenberg-mobile-pr-in-wpandroid)
+    - [How to share WIP WPAndroid APK](how-to-share-wip-wpandroid-apk)
+    - [How to checkout gutenberg-mobile to deployed binary version](#how-to-checkout-gutenberg-mobile-to-deployed-binary-version)
+
+---
+
+### Work with local gutenberg-mobile checkout
+
+- Make sure you run `npm install` or `npm ci`
+- Copy [local-builds.gradle-example](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/local-builds.gradle-example) renaming it to `local-builds.gradle`
+- Update `localGutenbergMobilePath`to your local `gutenberg-mobile`checkout
+- Run metro server with `npm run start:reset`
```

Comment:
I find these instructions a bit hard to follow because we don't say _where_ to run each command. Some should be run in `mobile-gutenberg/`, but others happen in the WPAndroid root (i.e., the copying and renaming of `local-builds.gradle-example` which doesn't say where the file should be copied to).

Reply:
One way to help this be clearer might be to group the `npm` steps together and the `WPAndroid` steps together instead of having `npm` -> `WPAndroid` -> `npm`. Just a thought.

Reply:
I've reworked this in 6a5ca7effc502ea798275d8ced92a54755d14942, let me know if that's a bit clearer.

Reply:
> One way to help this be clearer might be to group the npm steps together and the WPAndroid steps together instead of having npm -> WPAndroid -> npm. Just a thought.

I actually missed this comment while working on your feedback. So, I am glad I arrived at the same approach :)

Reply:
![](https://media1.giphy.com/media/7l5al5QCZRMLIPW9Cd/giphy.gif)